cns/service/main.go

```diff
@@ -424,6 +425,14 @@ func main() {
 
 	// Start CNS.
 	if httpRestService != nil {
+		if(cnsconfig.UseHttps){
+			config.TlsSettings = localtls.TlsSettings{
+				TlsCertificateSubjectName: cnsconfig.TlsSubjectName,
```

Comment:
It will fail as it stands if it goes into the windows implementation and is missing a subject name or goes into the file retriever implementation and doesn't specify cert path.